work_test.go

```diff
@@ -238,21 +239,19 @@ func TestLockJobAdvisoryRace(t *testing.T) {
 	getBackendID := func(conn *pgx.Conn) int32 {
 		var backendID int32
 		err := conn.QueryRow(`
-			SELECT backendid
-			FROM pg_stat_get_backend_idset() psgb(backendid)
-			WHERE pg_stat_get_backend_pid(psgb.backendid) = pg_backend_pid()
+			SELECT pg_backend_pid()
 		`).Scan(&backendID)
 		if err != nil {
 			panic(err)
 		}
 		return backendID
 	}
-	waitUntilBackendIsWaiting := func (backendID int32, name string) {
+	waitUntilBackendIsWaiting := func(backendID int32, name string) {
 		conn := newConn()
 		i := 0
 		for {
 			var waiting bool
-			err := conn.QueryRow(`SELECT pg_stat_get_backend_waiting($1)`, backendID).Scan(&waiting)
+			err := conn.QueryRow(`SELECT wait_event is not null from pg_stat_activity where pid=$1`, backendID).Scan(&waiting)
```

Comment:
`pg_stat_get_backend_waiting` could return true if the process is waiting for a lock, false if it is not waiting or null if the process id is invalid. The test would panic when null is returned.

In the changed query `ErrNoRows` is possible if the backendPID does not exist, in that case is it not appropriate to panic ?